bundles/core/src/main/java/com/adobe/cq/wcm/core/components/commons/link/LinkConstants.java

```diff
@@ -0,0 +1,41 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2021 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.commons.link;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * Constants for link handling.
+ */
+@ProviderType
+public final class LinkConstants {
```

Comment:
I'm not a fan of having the constants in a different class, I'd move them to `Link`.

Reply:
sonarcube usually complaints about putting constants in interface, and i agree it's better to put them in separate classes.
think of NameConstants in the WCM API.

Reply:
we may want to introduce a central class for constants, e.g. `CoreCmpConstants`

Reply:
To me, constants such as property names are actually part of API so they should also be part of the interface that defines the API.

Reply:
I think many people view this as an anti-pattern because Bloch wrote:

Reply:
Moved constants in ccbfe54